### PR TITLE
fix(react-alert): add live region attributes based on intent

### PR DESCRIPTION
**Pull request by smhigley**
I've added a role of `status` or `alert` based on the intent. I've also added `aria-live="assertive"` to all alerts, since my assumption is that any message important enough to use a component like Alert should probably be announced when it appears. This overrides the default live region priority of `status` while keeping `status` semantics.

### CODE REVIEW

**Comment by fabricteam** [comment]
## 📊 Bundle size report

<details>
<summary>Unchanged fixtures</summary>

| Package & Exports                                                                                                                                                             |          Size (minified/GZIP) |
| ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------- | ----------------------------: |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/MultipleComponents.fixture.js'>react-components: Accordion, Button, FluentProvider, Image, Menu, Popover</abbr> | `181.141 kB`<br />`50.598 kB` |
| <samp>react-components</samp> <br /> <abbr title='bundle-size/ProviderAndTheme.fixture.js'>react-components: FluentProvider & webLightTheme</abbr>                            |  `34.078 kB`<br />`11.098 kB` |

</details>
<sub>🤖 This report was generated against <a href='https://github.com/microsoft/fluentui/commit/87f555dfcbe007e701f80b56e3c69c50141e5466'>87f555dfcbe007e701f80b56e3c69c50141e5466</a></sub>
<div id="bundleSizeComment9423"/>

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/Avatar-mount_base.html">1335</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/Avatar-mount.html">1297</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/Button-mount_base.html">764</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/Button-mount.html">778</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProvider-mount_base.html">2322</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProvider-mount.html">2326</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProviderWithTheme-mount_base.html">407</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProviderWithTheme-mount.html">414</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">366</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">337</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">438</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">428</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/MakeStyles-mount_base.html">2044</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/23284/perf-test-react-components/MakeStyles-mount.html">2015</a></td>
            <td>50000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

File: packages/react-components/react-alert/src/components/Alert/useAlert.tsx
Comment (smhigley) [reply packages/react-components/react-alert/src/components/Alert/useAlert.tsx:57]: @rohitpagariya looks like the figma still needs to be updated. I'd prefer not to hold up the PR based on that, though. Are you OK with the change going in while design works on the updating the spec?

Comment (rohitpagariya) [reply packages/react-components/react-alert/src/components/Alert/useAlert.tsx:57]: Yes, please go ahead and merge the PR :)

Comment (jurokapsiar) [reply packages/react-components/react-alert/src/components/Alert/useAlert.tsx:57]: what is the behavior for assertive with role='status' - is that being considered by the screen readers? As far as I remember, alert is by default assertive. Do we need to make status assertive as well?

Comment (jurokapsiar) [reply packages/react-components/react-alert/src/components/Alert/useAlert.tsx:57]: my worry is that if this ends up being used for confirmations (something like 'logs uploaded', making it assertive might distract the user from what they are actually doing. I would much rather make it as an opt-in on the component level and let the product decide.

As far as I can see from the docs site, this is currently not the component that would be managing the toast nature, this appears to be just the content of the toast - @rohitpagariya can you please confirm if this is the plan?

**Review by rohitpagariya — APPROVED** [review]

File: packages/react-components/react-alert/src/components/Alert/useAlert.tsx
Comment (smhigley) [reply packages/react-components/react-alert/src/components/Alert/useAlert.tsx:57]: Even for something like "logs updated", I think `aria-live="assertive"` would be the better choice. Unless there are frequent interruptions, the feedback I've heard is that success/confirmation messages as assertive is a good thing.

The use case where I would use `aria-live="polite"` is more when something fully doesn't matter whether the user notices it or not. For a default, based on the design of Toast, I think polite isn't a good fit. As @khmakoto mentioned, if an author has a specific use case where the toasts are frequent or particularly unimportant, they can customize the `aria-live` value (though in that case the visual distraction might be its own separate issue).

Comment (rohitpagariya) [reply packages/react-components/react-alert/src/components/Alert/useAlert.tsx:57]: With the toasts feature, Teams will have two temporal notification systems - banners & toasts. It's not clear to me when an app should use banners vs toasts. I will follow up with design / PM team on clear & written guidance.

Today, banners in Teams use role="alert" & "aria-live=polite" by default. And these aria attributes don't change based on the intent. 

To Juraj's question, the current plan is to use Alert to render the content of the toast. I have already built a ToastProvider component to manage the incoming & outgoing toasts and ToastRenderer component to handle animations, auto dismiss & rendering the toast data via the Alert component. 

The v0 Alert component also sets role="alert" & "aria-live=polite" by default

Comment (diana-hsu) [reply packages/react-components/react-alert/src/components/Alert/useAlert.tsx:57]: Toasts are meant to be used for transient notifications, typically in response to an action the user just took. And so the most typical use of the toast will be something like "download completed" or "logs uploaded," thus polite is appropriate because one of the top complaints from screen readers is too much information and constant distractions. Let's cut down on those distractions where we can -- in fact, Brett said that many screen reader users even pay for a Teams add-on which turns off most of the Teams notifications from being narrated (!).

If there is something that the user needs to know, such as their download couldn't be completed, that can be an error and given priority so that users can take action.

The decision tree for banner vs toast usage can be found here: https://www.figma.com/file/HQ4nlkneIZvWUFzehrQoEL/Toast?node-id=2075%3A221556

**Review by ling1726 — APPROVED** [review]

**Comment by ling1726** [comment]
Approving to unblock, but would be good to get consensus on the default role

**Comment by diana-hsu** [comment]
> Approving to unblock, but would be good to get consensus on the default role

@ling1726 Brett confirmed that default should be polite

**Pull request merged**
